pkg/utils/json.go

```diff
@@ -0,0 +1,52 @@
+package utils
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+)
+
+const (
+	// JSONDoc type for json files
+	JSONDoc = "json"
+)
+
+// LoadJSON loads a JSON file into an IacDocument struct
+func LoadJSON(filePath string) ([]*IacDocument, error) {
+	iacDocumentList := make([]*IacDocument, 0, 1)
+
+	// First pass determines line number data
+	currentLineNumber := 1
+	{ // Limit the scope for Close()
+		file, err := os.Open(filePath)
+		if err != nil {
+			return iacDocumentList, err
+		}
+		defer file.Close()
+
+		scanner := bufio.NewScanner(file)
+		for scanner.Scan() {
+			currentLineNumber++
+		}
+
+		if err = scanner.Err(); err != nil {
+			return iacDocumentList, err
+		}
+	}
+
+	// Second pass extracts raw data
+	fileBytes, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return iacDocumentList, err
+	}
+
+	iacDocumentList = append(iacDocumentList, &IacDocument{
+		Type:      JSONDoc,
+		StartLine: 1,
```

Comment:
Can we have multiple k8s resources defined in the same json? then in that case we may have to take care of the `StartLine`

Reply:
As far as I can tell, only yaml has the multi docs feature, which is the only reason we have to support multiple resources there. And from the schema, it appears that we would only have one definition per "document", so I believe the startline will always need to be 1 for JSON. What we might want to add though is the end line for resources in our results output